File: backend/src/app/schemas/github_import.py
import uuid

from pydantic import BaseModel


class AIGeneratedProject(BaseModel):
    repo_id: uuid.UUID
    title: str
    description: str | None = None
    markdown_content: str | None = None
    github_url: str


class ImportAnalyzeRequest(BaseModel):
    repo_ids: list[uuid.UUID]
    facet_id: uuid.UUID


class ImportAnalyzeResponse(BaseModel):
    projects: list[AIGeneratedProject]


class ImportConfirmRequest(BaseModel):
    facet_id: uuid.UUID
    projects: list[AIGeneratedProject]


class ImportConfirmResponse(BaseModel):
    message: str
    count: int
    project_ids: list[uuid.UUID]

File: frontend/src/lib/api/github-import.ts
import { apiClient } from "@/lib/api-client";
import { getToken } from "@/lib/auth/token";

export interface AIGeneratedProject {
  repo_id: string;
  title: string;
  description: string | null;
  markdown_content: string | null;
  github_url: string;
}

export interface ImportAnalyzeResponse {
  projects: AIGeneratedProject[];
}

export interface ImportConfirmResponse {
  message: string;
  count: number;
  project_ids: string[];
}

function getAuthHeaders(): Record<string, string> {
  const token = typeof window !== "undefined" ? getToken() : null;
  return token ? { Authorization: `Bearer ${token}` } : {};
}

function withAuth(opts: Record<string, unknown> = {}): Record<string, unknown> {
  return { ...opts, headers: { ...getAuthHeaders(), ...(opts.headers as Record<string, string> ?? {}) } };
}

export const githubImportApi = {
  analyze: (repoIds: string[], facetId: string) =>
    apiClient.post<ImportAnalyzeResponse>("/api/github/import/analyze", { repo_ids: repoIds, facet_id: facetId }, withAuth()),

  confirm: (facetId: string, projects: AIGeneratedProject[]) =>
    apiClient.post<ImportConfirmResponse>("/api/github/import/confirm", { facet_id: facetId, projects }, withAuth()),
};
